pkg/test/e2e/utils.go

```diff
@@ -0,0 +1,248 @@
+				return ctx, fmt.Errorf("delete namespace func: %w", err)
+			}
+
+			err = wait.For(conditions.New(client.Resources()).ResourceDeleted(namespace), wait.WithTimeout(time.Minute*1))
+			if err != nil {
+				return ctx, fmt.Errorf("waiting for delete namespace func: %w", err)
+			}
+		}
+
+		//create namespace
+		namespace = &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: name}}
+		if err = client.Resources().Create(ctx, namespace); err != nil {
+			return ctx, fmt.Errorf("create namespace func: %w", err)
+		}
+
+		fmt.Printf("====> Done.\n")
+		cfg.WithNamespace(name)
+		return context.WithValue(ctx, namespaceContextKey(name), namespace), err
+	}
+}
+
+func e2eDeployEnvironmentResources(yamlFiles []string, namespace string) env.Func {
+	return func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
+		fmt.Printf("====> Deploying k8s resources from - %v\n", yamlFiles)
+		yaml := ""
+		for _, file := range yamlFiles {
+			fileYaml, err := ioutil.ReadFile(file)
+			if err != nil {
+				return ctx, fmt.Errorf("e2eDeployEnvironmentResources - ReadFile: %w", err)
+			}
+
+			yaml += "\n---\n" + string(fileYaml)
+		}
+
+		resources := parseK8sYaml(string(yaml))
+		client, err := cfg.NewClient()
+		if err != nil {
+			return ctx, fmt.Errorf("e2eDeployEnvironmentResources - NewClient: %w", err)
+		}
+
+		for _, resource := range resources {
+			k8sResource := resource.(k8s.Object)
+			k8sResource.SetNamespace(namespace)
+
+			// Recreate resource
+			// Note: Skip if resource do not exist, we create
+			_ = client.Resources().Delete(ctx, k8sResource)
+			err = client.Resources().Create(ctx, k8sResource)
+			if err != nil {
+				return ctx, fmt.Errorf("create resource : %w", err)
+			}
+		}
+		fmt.Printf("====> Done.\n")
+		return ctx, nil
+	}
+}
+
+func parseK8sYaml(yaml string) []runtime.Object {
+	acceptedK8sTypes := regexp.MustCompile(`(ConfigMap|Service|Deployment|ServiceAccount|ClusterRole|ClusterRoleBinding|Secret)`)
+	sepYamlFiles := strings.Split(yaml, "---")
+	resources := make([]runtime.Object, 0, len(sepYamlFiles))
+	for _, f := range sepYamlFiles {
+		if f == "\n" || f == "" {
+			// ignore empty cases
+			continue
+		}
+
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		resource, groupVersionKind, err := decode([]byte(f), nil, nil)
+
+		if err != nil {
+			continue
+		}
+
+		if !acceptedK8sTypes.MatchString(groupVersionKind.Kind) {
+		} else {
+			resources = append(resources, resource)
+		}
+	}
+	return resources
+}
+
+func GetConfigSet(confFileName string) (*corev1client.CoreV1Client, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", confFileName)
+	if err != nil {
+		return nil, err
+	}
+
+	clientSet, err := corev1client.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientSet, err
+}
+
+func LogsFromPods(pods corev1.PodList, clientSet *corev1client.CoreV1Client, namespace string) string {
+	logs := ""
```

Comment:
Consider using `strings.Builder`. It should be more efficient.
https://www.calhoun.io/concatenating-and-building-strings-in-go/

Reply:
updated :-)